chat-android/src/main/java/com/ably/chat/Message.kt

```diff
@@ -71,37 +71,62 @@ public data class Message(
      * Do not use the headers for authoritative information. There is no server-side
      * validation. When reading the headers treat them like user input.
      */
-    val headers: MessageHeaders,
+    public val headers: MessageHeaders
 
     /**
      * The latest action of the message. This can be used to determine if the message was created, updated, or deleted.
      * Spec: CHA-M10
      */
-    val action: MessageAction,
+    public val action: MessageAction
 
     /**
      * A unique identifier for the latest version of this message.
      * Spec: CHA-M10a
      */
-    val version: String,
+    public val version: String
 
     /**
      * The timestamp at which this version was updated, deleted, or created.
      */
-    val timestamp: Long,
+    public val timestamp: Long
 
     /**
      * The details of the operation that modified the message. This is only set for update and delete actions. It contains
      * information about the operation: the clientId of the user who performed the operation, a description, and metadata.
      */
-    val operation: Message.Operation? = null,
-)
+    public val operation: Operation?
+}
 
-internal fun buildMessageOperation(jsonObject: JsonObject?): Message.Operation? {
+public fun Message.copy(
+    text: String = this.text,
+    headers: MessageHeaders = this.headers,
+    metadata: MessageMetadata = this.metadata,
+): Message =
```

Comment:
Signature should be

```suggestion
public fun Message.copy(
    text: String,
    headers: MessageHeaders = this.headers,
    metadata: MessageMetadata = this.metadata,
): Message =
```
`text` is mandatorysince we are using `copy` method specifically for `messages.update`. 
https://ably.com/docs/chat/rooms/messages?lang=swift#update

Reply:
Since, `copy` is default internal method in the case of `kotlin` ( doesn't exist in `swift` and `js` ), if want to generalize it, then it should be renamed to something else.

Reply:
> text is mandatory and shouldn't be inferred from underlying object

not sure this is true, don't see anything in the spec about this

>  copy is default internal method in the case of kotlin

copy is not default, nor internal, it simply autogenerated for data classes, and semantic is exactly the same

Reply:
Yeah, that's what I meant. In case of `swift` it's not autogenerated, hence method signature is different. 
And `copy` method is only used in the context of `messages.update`. 
So, it would make sense to define context-specific method signature.

Reply:
I am not sure what you are suggesting, do you think `copy` is bad name and we should rename it? In this case what name do you think fits better? (I personally very much like `copy` because it's something Kotlin user would expect for something like this, it works exactly like data classes `copy`, it creates new instance instead of mutate current one, you can specify arguments you want, there is no confusion here). Or you don't like that text is optional? I believe sdk users should be able to update headers or metadata without changing the text, there is no such requirements in the spec and js implementation.

Reply:
[Based on the discussion](https://ably-real-time.slack.com/archives/C03JDBVM5MY/p1742836867116739), we are good 👍